src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3693,6 +3720,75 @@ void Lowering::LowerHWIntrinsicToScalar(GenTreeHWIntrinsic* node)
         LowerNode(cast);
     }
 }
+
+//----------------------------------------------------------------------------------------------
+// Lowering::LowerAndOpToResetLowestSetBit: Lowers a tree AND(X, ADD(X, -1) to HWIntrinsic::ResetLowestSetBit
+//
+// Arguments:
+//    andNode - GentreePtr of binOp
+//
+// Return Value:
+//    Returns the replacement node if one is created else nullptr indicating no replacement
+//
+// Assumptions:
+//    andNode is GT_AND and (TYP_INT or TYP_LONG)
+//
```

Comment:
```suggestion
//----------------------------------------------------------------------------------------------
// Lowering::LowerAndOpToResetLowestSetBit: Optimize AND(X, ADD(X, -1) to HWIntrinsic::ResetLowestSetBit
//
// Arguments:
//    andNode - GT_AND node of integral type
//
// Return Value:
//    Returns the next node to lower if the optimization was successful, "nullptr" otherwise.
//
```
(I would consider changing the name from `Lower` to `Optimize` too)

Reply:
I've removed assumptions. Though it would make sense to me that if we're asserting something at the beginning or end of the method that those are sensible things to document as pre and post conditions.

Reply:
I agree, and I didn't mean to drop that part, just move it to the argument description: (right now it still reads `GentreePtr of binOp`, where `GentreePtr` has been [not a thing](https://github.com/dotnet/coreclr/pull/16027) for quite some time).

Reply:
Fixed.